File: backend/src/main/java/com/ll/hotel/domain/member/member/repository/MemberRepository.java
package com.ll.hotel.domain.member.member.repository;

import com.ll.hotel.domain.member.member.entity.Member;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface MemberRepository extends JpaRepository<Member, Long> {
    Optional<Member> findByMemberEmail(String memberEmail);
    Optional<Member> findByMemberName(String nickname);
    boolean existsByMemberEmail(String memberEmail);
}

File: backend/src/main/java/com/ll/hotel/domain/member/member/repository/MemberRepository.kt
package com.ll.hotel.domain.member.member.repository

import com.ll.hotel.domain.member.member.entity.Member
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.Optional

@Repository
interface MemberRepository : JpaRepository<Member, Long> {
    fun findByMemberEmail(memberEmail: String): Optional<Member>
    fun findByMemberName(nickname: String): Optional<Member>
    fun existsByMemberEmail(memberEmail: String): Boolean
} 